plugins/catalog-backend/src/ingestion/processors/codeowners/resolve.ts

```diff
@@ -18,6 +18,10 @@ import * as codeowners from 'codeowners-utils';
 import { CodeOwnersEntry } from 'codeowners-utils';
 import { filter, get, head, pipe, reverse } from 'lodash/fp';
 
+const USER_PATTERN = /^@.*/;
+const GROUP_PATTERN = /^@.*\/.*/;
+const EMAIL_PATTERN = /^.*@.*\..*$/;
```

Comment:
Uff, these regexes are pretty loose and become order dependent ... but that's not your fault :) Just noting that some day, it might make sense to break out codeowners as its own `@backstage/plugin-catalog-module-codeowners` and let users pass in a translation function to the constructor of the processor, that translates from an entry type to `EntityName`. Like

```ts
type CodeownersEntry =
  | { type: 'user', rawEntry: string, username: string }
  | { type: 'team', rawEntry: string, org: string, teamName: string }
  | { type: 'email', rawEntry: string };
```

or something like that. Some other day! But that would let them deal with things like the ugly email substring thingy in a custom way (maybe searching for a matching LDAP user by attribute or whatever...)

Reply:
Yeah flexibility like that would be great and for processors in general. It's perhaps a slightly more complicated use case but similar functionality could be useful in the github-org processor. I recall a user asked if they could associate documentation with a group but since the org processor was creating the entity we didn't have a point to do it. Similarly we couldn't configure entity links either.  Now I think there was talk of making documentation a kind (?) in which case a relation could do. And thinking out loud another custom processor could always add the required fields.

Reply:
Oh yes! You'll see that we have extracted a few things already (like the ldap org processor) and that work should definitely continue. The catalog should have fewer things builtin, and require a little bit more manual setup. I think that's fine, and will have great benefits for flexibility.

If you're eager to get something like this done for a specific processor, do make a contribution if you have the time to! We're gonna be very starved for resources over the next month or so when everybody goes on vacation :) Otherwise it'll probably happen a bit later

Reply:
Awesome, we're quite keen to make more contributions in general be it in the catalog or elsewhere. We should conversely have more time to do that 😄